src/Type/Constant/ConstantArrayType.php

```diff
@@ -386,6 +387,12 @@ public function findTypeAndMethodName(): ?ConstantArrayTypeAndMethod
 			if ($this->isOptionalKey(0) || $this->isOptionalKey(1)) {
 				$has = $has->and(TrinaryLogic::createMaybe());
 			}
+			if (
+				$classOrObject instanceof ConstantStringType &&
+				!$type->getMethod($method->getValue(), new OutOfClassScope())->isStatic()
+			) {
+				return null;
+			}
```

Comment:
This one feels tricky and awkward. We know a class is invoked statically when the type is `ConstantStringType` but we cannot determine that from the current type – we need a typehint on the original object and then fetch the reflection via the determined type.

Reply:
Shoot. I think this also broke the CI with analysis here:

```
in_array($methodReflection->getDeclaringClass()->getName(), [DateTime::class, DateTimeImmutable::class], true)
```

Reply:
1) I recommend you to create a variable `$isClassString` and set it to true in the `if ($classOrObject instanceof ConstantStringType) {` and ` elseif ($classOrObject instanceof GenericClassStringType) {`.
2) Before calling getMethod you need to check that `$has` is `yes()`.

Otherwise I think it's fine.

Reply:
I also need to add more strict checks. 

This was failing as well:

```
private const MAPPING = [\stdClass::class, RandomAbstractClass::class];
```

Or maybe my changes uncovered a silent bug? 

```
		$has = $type->hasMethod($method->getValue());
		if (!$has->no()) {
```

That should have been false because it's an array of classes? Or was it always reporting `maybe` which triggered `$has = $has->and(TrinaryLogic::createMaybe());` and just carried on?

Either way, next week when I'm back on this I'll dive in deeper and add your suggestion